plugins/randomforest_procedure.cc

```diff
@@ -0,0 +1,339 @@
+/** randomforest_procedure.cc                                             -*- C++ -*-
+    Mathieu Marquis Bolduc, 11 Mars 2016
+    Copyright (c) 2016 Datacratic Inc.  All rights reserved.
+
+    This file is part of MLDB. Copyright 2016 Datacratic. All rights reserved.
+
+    Procedure to train a random forest binary classifier.
+*/
+
+#include "randomforest_procedure.h"
+#include "mldb/arch/bit_range_ops.h"
+#include "mldb/arch/bitops.h"
+#include "mldb/arch/timers.h"
+#include "mldb/jml/utils/profile.h"
+#include "mldb/ml/randomforest.h"
+#include "mldb/ml/value_descriptions.h"
+#include "mldb/plugins/sql_expression_extractors.h"
+#include "mldb/plugins/classifier.h"
+#include "mldb/rest/in_process_rest_connection.h"
+#include "mldb/server/bound_queries.h"
+#include "mldb/server/mldb_server.h"
+#include "mldb/types/any_impl.h"
+#include "mldb/types/basic_value_descriptions.h"
+#include "mldb/types/set_description.h"
+#include "mldb/vfs/fs_utils.h"
+
+#include <boost/random/mersenne_twister.hpp>
+#include <boost/random/uniform_int.hpp>
+#include <boost/random/uniform_01.hpp>
+
+using namespace std;
+using namespace ML;
+
+namespace Datacratic {
+namespace MLDB {
+
+DEFINE_STRUCTURE_DESCRIPTION(RandomForestProcedureConfig);
+
+RandomForestProcedureConfigDescription::
+RandomForestProcedureConfigDescription()
+{
+    addField("trainingData", &RandomForestProcedureConfig::trainingData,
+             "Specification of the data for input to the classifier procedure. "
+             "The select expression must contain these two sub-expressions: one row expression "
+             "to identify the features on which to train and one scalar expression "
+             "to identify the label.  The type of the label expression must match "
+             "that of the classifier mode: a boolean (0 or 1) for `boolean` mode; "
```

Comment:
The last part about the mode does not apply yet, right?  If so, we should remove it.